src/js/Calculator.js

```diff
@@ -0,0 +1,152 @@
+import Operator from "./domain/Operator.js";
+import CalculatorView from "./view/CalculatorView.js";
+import {
+  STATE_KEY,
+  OPERATOR_SYMBOL,
+  MESSAGE,
+  MAX_OPERAND_LENGTH,
+} from "./constant/index.js";
+
+const initState = {
+  total: 0,
+  operator: "",
+  leftOperand: "",
+  rightOperand: "",
+};
+
+class Calculator {
+  constructor({ $total, $digits, $modifiers, $operations }) {
+    this.view = new CalculatorView({ $target: $total });
+    this.$total = $total;
+    this.$digits = $digits;
+    this.$modifiers = $modifiers;
+    this.$operations = $operations;
+    this.state = {};
+    this.init();
+  }
+
+  init() {
+    this.setInitState();
+    this.handleDigitClick();
+    this.handleModifierClick();
+    this.handleOperatorClick();
+  }
+
+  setInitState() {
+    this.state = { ...initState };
+  }
+
+  handleDigitClick() {
+    this.$digits.addEventListener("click", ({ target }) => {
+      const digit = target.textContent;
+      this.setOperandByOperator(digit);
+      this.view.renderStatus(this.state);
+    });
+  }
+
+  handleModifierClick() {
+    this.$modifiers.addEventListener("click", () => {
+      this.setInitState();
+      this.view.renderTotal(this.state.total);
+    });
+  }
+
+  handleOperatorClick() {
+    this.$operations.addEventListener("click", ({ target }) => {
+      if (!this.validateOperateOrder()) return;
+
+      const symbol = target.textContent;
+      const symbolKey = target.dataset.symbol.toUpperCase();
+
+      if (this.isResultButtonClicked(symbol)) {
+        this.calculate();
+        this.setTotalToOperand(this.state.total);
+        this.view.renderTotal(this.state.total);
+        return;
+      }
+
+      this.setState({ key: STATE_KEY.OPERATOR, newState: symbolKey });
+      this.view.renderStatus(this.state);
+    });
+  }
+
+  isResultButtonClicked(symbol) {
+    return symbol === OPERATOR_SYMBOL.RESULT;
+  }
+
+  calculate() {
+    const { operator, leftOperand, rightOperand } = this.state;
+    const total = Operator[`${operator}`](
+      Number(leftOperand),
+      Number(rightOperand)
+    );
```

Comment:
```suggestion
  calculate() {
    const { operator, leftOperand, rightOperand } = this.state;
    const total = Operator[operator](
      Number(leftOperand),
      Number(rightOperand)
    );
```

템플릿리터럴 사용안해두 될것 같아요!

(마이너) 객체 indexing 할때 undefined 반환되는 경우에 에러핸들링해주면 더 좋지 않을까 생각해봅니다 ㅎㅎ. undefined is not a funciton 뜨면 힘 빠지더라구요

Reply:
캬~~~`undefined 반환되는 경우에 에러핸들링` 까지 생각 못했는데 알려주셔서 감사해요 ㅠㅠ